docs/Integration_Guide.md

```diff
@@ -0,0 +1,115 @@
+The Process of Spherex Protection into the smart contracts is usually done (at least at this point) by the tech team at Spherex.
+
+That being said, and because the developer of the smart contracts will always know it better than we do, it is important for you as the developer of the protected contract to undestand the process of integration and the various aspects of that process.
+
+- [Protection  - the big picture](#protection----the-big-picture)
+- [Some terms and defenitions](#some-terms-and-defenitions)
+    - [Contracts in the system](#contracts-in-the-system)
+    - [Entities and addresses](#entities-and-addresses)
+- [Integration](#integration)
+  - [Inline vs Proxy](#inline-vs-proxy)
+  - [Configuration](#configuration)
+  - [Deployment](#deployment)
+    - [Special cases](#special-cases)
+- [What Should you do?](#what-should-you-do)
+  - [Roles and Permissions](#roles-and-permissions)
+  - [For each of you'r contracts:](#for-each-of-your-contracts)
+    - [For immutable contracts (inline protection):](#for-immutable-contracts-inline-protection)
+    - [For mutable contracts (proxy protection):](#for-mutable-contracts-proxy-protection)
+    - [In the deployment Scripts](#in-the-deployment-scripts)
+
+
+## Protection  - the big picture
+- Spherex protection works by identifiying suspicious and malicious behavior patterns of a transaction as it is being executed by the protocol (between any number of contracts)
+- The granularity of the protection is **functions** - meaning the smallest unit of code we track behaviour of is a function.
+  - The data collected and used for each function in each protocol may change, depending on various factors (gas cost in the chain, past history, client requests, etc.)
+- `pure` and `view` functions are not protected, as they do not change the state of the contract , and therefore do not require protection.
+
+## Some terms and defenitions
+
+#### Contracts in the system
+
+- **Protected Contract**: The smart contract that is being protected by Spherex.
+- **Spherex Engine**: This is a smart contract written by spherex that acts as *the brain* of the protection system.
+  - The protected contracts in the protocol communicate with the Spherex Engine to perform various operations (send information and get approval for the current state of the transaction).
```

Comment:
There's only single operation being performed - revert or keep going

Reply:
how about now?

Reply:
The protected contracts don't revert